openpype/hosts/max/plugins/publish/collect_tycache_attributes.py

```diff
@@ -0,0 +1,79 @@
+import pyblish.api
+
+from openpype.lib import EnumDef, TextDef
+from openpype.pipeline.publish import OpenPypePyblishPluginMixin
+
+
+class CollectTyCacheData(pyblish.api.InstancePlugin,
+                         OpenPypePyblishPluginMixin):
+    """Collect Review Data for Preview Animation"""
+
+    order = pyblish.api.CollectorOrder + 0.02
+    label = "Collect tyCache attribute Data"
+    hosts = ['max']
+    families = ["tycache"]
+
+    def process(self, instance):
+        all_tyc_attributes_dict = {}
+        attr_values = self.get_attr_values_from_data(instance.data)
+        tycache_boolean_attributes = attr_values.get("all_tyc_attrs")
+        if tycache_boolean_attributes:
+            for attrs in tycache_boolean_attributes:
+                all_tyc_attributes_dict[attrs] = True
+        tyc_layer_attr = attr_values.get("tycache_layer")
+        if tyc_layer_attr:
+            all_tyc_attributes_dict["tycacheLayer"] = (
+                tyc_layer_attr)
+        tyc_objname_attr = attr_values.get("tycache_objname")
+        if tyc_objname_attr:
+            all_tyc_attributes_dict["tycache_objname"] = (
+                tyc_objname_attr)
+        self.log.debug(
+            f"Found tycache attributes: {all_tyc_attributes_dict}")
```

Comment:
Other than reporting the found attributes does this do ANYTHING with that data? It doesn't store it on the instance or context?

Also, `all_tyc_attributes_dict` is way verbose making this code hard to read:

```suggestion
    def process(self, instance):
        attr_values = self.get_attr_values_from_data(instance.data)

        attributes = {}
        for attr_key in attr_values.get("all_tyc_attrs", []):
            attributes[attr_key] = True
            
        tyc_layer_attr = attr_values.get("tycache_layer")
        if tyc_layer_attr:
            attributes["tycacheLayer"] = tyc_layer_attr
            
        tyc_objname_attr = attr_values.get("tycache_objname")
        if tyc_objname_attr:
            attributes["tycache_objname"] = tyc_objname_attr
        
        # TODO: Do we need this data somewhere?
        instance.data["storeWhereverYouWant"] = attributes
        self.log.debug(
            f"Found tycache attributes: {attributes}"
        )
```

also, what's the reason we're converting attribute `tycache_layer` to `tycacheLayer` from attribute definition to the attributes dict but `tycache_objname` remains `tycache_objname`? I feel this is a bit of a code style issue. If the attribute def keys would match the names you want to store them in the dict with you could just do:

```suggestion
    def process(self, instance):
        attr_values = self.get_attr_values_from_data(instance.data)

        attributes = {}
        for attr_key in attr_values.get("all_tyc_attrs", []):
            attributes[attr_key] = True
        
        for key in ["tycacheLayer", "tycacheObjName"]:
            if key in attr_values:
                attributes[key] = attr_values[key]
        
        # TODO: Do we need this data somewhere?
        instance.data["storeWhereverYouWant"] = attributes
        self.log.debug(
            f"Found tycache attributes: {attributes}"
        )
```

However, you might be better off always SETTING the 'default' value for an option even if `""` for consistency of the `instance.data`
```suggestion
    def process(self, instance):
        attr_values = self.get_attr_values_from_data(instance.data)

        attributes = {}
        for attr_key in attr_values.get("all_tyc_attrs", []):
            attributes[attr_key] = True
        
        for key in ["tycacheLayer", "tycacheObjName"]:
            attributes[key] = attr_values.get(key, "")
        
        # TODO: Do we need this data somewhere?
        instance.data["storeWhereverYouWant"] = attributes
        self.log.debug(
            f"Found tycache attributes: {attributes}"
        )
```

Reply:
I would try to see if it pass the validation tmr